text/0021-accessibility.md

```diff
@@ -0,0 +1,485 @@
+        user_interface.operate(&renderer, operation.as_mut());
+
+        match operation.finish() {
+            operation::Outcome::None => {}
+            operation::Outcome::Some(message) => match message {
+                operation::OperationOutputWrapper::Message(
+                    _,
+                ) => {
+                    unimplemented!();
+                }
+                operation::OperationOutputWrapper::Id(id) => {
+                    focus = Some(A11yId::from(id));
+                }
+            },
+            operation::Outcome::Chain(next) => {
+                current_operation = Some(Box::new(
+                    OperationWrapper::Wrapper(next),
+                ));
+            }
+        }
+    }
+
+    adapter.update(TreeUpdate {
+        nodes: window_tree.into(),
+        tree: Some(tree),
+        focus: focus.map(|id| id.into()),
+    });
+}
+```
+- reset widget Id counter before building the UserInterface
+
+See https://github.com/wash2/iced for more implementation details
+
+## Drawbacks
+
+Drawbacks of this proposal are that the accessibility tree must be rebuilt every redraw. Accesskit also is still missing some widget implementations, though I'm not sure what they are. Lastly, the way focus is recalculated every redraw after accessibility is enabled may cause a performance hit depending on the application. 
+
+
+## Rationale and alternatives
+
+- Why is this design the best in the space of possible designs?
+
+This design is currently the only design, but it aims to neatly fit into the existing architecture and support incremental implementation.
+
+- What other designs have been considered and what is the rationale for not choosing them?
+
+I believe other designs might use a persistent widget tree and a diffing algorithm, which don't currently exist in Iced. Another option might use some global state store for faster lookups, though I'm still not entirely sure it's a major issue to begin with.
+
+- What is the impact of not doing this?
+
+Iced would have to implement Accessibility some other way. If Iced does not support Accessibility, developers may choose another toolkit.
+
+## Prior art
+
+Egui supports accessibility via Accesskit. The Egui widget implementations could probably be used as a reference when implementing this proposal.
+
+
+## Unresolved questions
+
+- Maybe there should be a new Id variant for Groups of Ids in a single widget?
+- Maybe there should be more helper methods for the different widget types?
+- What related issues do you consider out of scope for this RFC that could be addressed in the future independently of the solution that comes out of this RFC?
+
+Actual Widget implementations are somewhat out of scope of this RFC. Of course I hope that most widgets can be easily implemented with the skeleton proposed here, but it may become useful to define templates for different `Node`s or more helpers for building `A11yTree`s
```

Comment:
I think you wanted this to replace the last item in the list above?

Reply:
yes :sweat_smile: